### PR TITLE
Feature/local references

### PR DESCRIPTION
**Pull request by survivorbat**
- Add local references to schema and populate them on scheyaml traversal. 
- Add error return value to JSONSchema.ScheYAML method

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -39,6 +39,11 @@ type Config struct {
 	// LineLength prevents descriptions and unreasonably long lines. Can be disabled
 	// completely by setting it to 0.
 	LineLength uint
+
+	// rootSchema is used to keep track of the root of the schema we're traversing, in case
+	// we want to use lookups to find a $ref. Not exported as users aren't meant to tinker
+	// with this value yet.
+	rootSchema *JSONSchema
```
Comment (Emptyless) [inline config.go:46]: a $ref can also follow a filepath (file://) which is relative to some dir, how would such a ref be evaluated?

File: ref.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	case len(pathSegments) == 0:
+		return object, nil
+
+	// If a lone # is input, we return the object itself
+	case len(pathSegments) == 1 && pathSegments[0] == "#":
+		return object, nil
+
+	// If this is the first segment and it has a #, we recursively traverse the next segment
+	case len(pathSegments) > 1 && pathSegments[0] == "#":
+		return lookup(object, pathSegments[1:])
```
Comment (Emptyless) [inline ref.go:57]: subsequent passes won't start with #/ anymore, how would you know its still a local ref?

File: config.go (hunk 1 above)
Comment (survivorbat) [reply config.go:46]: Both remote and file references are currently not in scope for this change, and I might never add those as I don't have a use-case for it

File: ref.go (hunk 2 above)
Comment (survivorbat) [reply ref.go:57]: I'm not sure how that is relevant for this function in particular, this traverses in-memory maps primarily

**Review by Emptyless — APPROVED** [review]

**Pull request merged**